File: web/sdk/exemplos/02_listar_itens.js
/* Exemplo 2/10 — listar itens: uma página, todas as páginas (cursor embutido) e facetas. */
import { comSessao, preparar } from './_comum.js';

preparar('02_listar_itens', (ctx) => comSessao(ctx, 'sdk-js-exemplo-02', async (p) => {
  const pagina = await p.itens.listar({ limite: 5 });
  ctx.escrever(`primeira página: ${pagina.itens.length} de ${pagina.total} item(ns)`);
  for (const it of pagina.itens) ctx.escrever(`  ${it.tipo}  ${it.titulo}`);
  let n = 0;
  for await (const it of p.itens.todos({ limite: 5 })) { n += 1; if (n >= 50) break; }
  ctx.escrever(`todos(): ${n} item(ns) percorridos sem ler proximo_cursor`);
  const facetas = await p.itens.facetas();
  ctx.escrever(`facetas por tipo: ${facetas.tipo.map((f) => `${f.valor}=${f.n}`).join(', ')}`);
  return { total: pagina.total, percorridos: n };
}));

File: sdk/python/exemplos/02_listar_itens.py
"""Exemplo 2/10 — listar o catálogo (`GET /api/itens`), uma página e todas."""

from __future__ import annotations

from plat import Plataforma


def main(url: str, inquilino: str, login: str, senha: str) -> None:
    p = Plataforma.entrar(url, inquilino, login, senha, nome_token="sdk-python-exemplo-02")
    try:
        pagina = p.itens.listar(limite=5)
        assert set(pagina) >= {"total", "itens", "proximo_cursor"}
        assert isinstance(pagina["itens"], list)
        print(f"página 1: {len(pagina['itens'])} de {pagina['total']} item(ns)")

        # paginação embutida: itera o catálogo inteiro sem ler proximo_cursor na mão
        contagem = sum(1 for _ in p.itens.todos())
        assert contagem == pagina["total"]
        print(f"todos(): {contagem} item(ns) no total, batendo com a página 1")
    finally:
        p.tokens.revogar(p.token_id)
        p.fechar()


if __name__ == "__main__":
    from _ambiente import do_ambiente

    main(*do_ambiente())
